### PR TITLE
Fixes stasis bed runtimes

### PR DESCRIPTION
**Pull request by Archanial**
## About The Pull Request

This PR limits stasis-opcomputer interaction to humans and monkeys.
Also opcomputers get monkey data from optables.
## Why It's Good For The Game

Removes runtime when buckling something that is not human to stasis.

## Changelog
:cl:
tweak: Operation computers only recognize humans or monkeys as patients.
/:cl:

### CODE REVIEW

File: code/game/machinery/stasis.dm [hunk 1]
```diff
@@ -127,7 +127,7 @@
 	update_icon()
 
 /obj/machinery/stasis/proc/check_patient()
```
Comment (spookydonut) [inline code/game/machinery/stasis.dm:129]: what kind of shittery is this proc, just replace any uses of this with `if(ishuman(occupant) ||  ismonkey(occupant))`

Comment (spookydonut) [reply code/game/machinery/stasis.dm:129]: yep literally only /obj/machinery/computer/operating/ui_data(mob/user) uses this

File: code/game/objects/structures/tables_racks.dm [hunk 2]
```diff
@@ -544,11 +544,12 @@
 	check_patient()
 
 /obj/structure/table/optable/proc/check_patient()
-	var/mob/living/carbon/human/M = locate(/mob/living/carbon/human, loc)
+	var/mob/living/carbon/M = locate(/mob/living/carbon, loc)
```
Comment (spookydonut) [inline code/game/objects/structures/tables_racks.dm:547]:
```suggestion
	var/mob/living/carbon/M = locate(/mob/living/carbon) in loc

```

File: code/game/machinery/stasis.dm (hunk 1 above)
Comment (Archanial) [reply code/game/machinery/stasis.dm:129]: Yeah, you're right

File: code/game/objects/structures/tables_racks.dm (hunk 2 above)
Comment (Archanial) [reply code/game/objects/structures/tables_racks.dm:547]: What's the difference?

Comment (spookydonut) [reply code/game/objects/structures/tables_racks.dm]: one is correct and one isnt

Comment (spookydonut) [reply code/game/objects/structures/tables_racks.dm]: there's no two argument version of locate()

Comment (spookydonut) [reply code/game/objects/structures/tables_racks.dm]: >     locate(Type) in Container 
    locate(X,Y,Z) 
    locate(Tag) 
    locate(TextRef)

Comment (Archanial) [reply code/game/objects/structures/tables_racks.dm]: What about something like: 
`locate(/obj/machinery/computer/operating, get_step(src, direction))`?

Comment (spookydonut) [reply code/game/objects/structures/tables_racks.dm]: There's no two argument version of `locate()`

**Comment by Archanial** [comment]
I've replaced all 2 argument locate() with locate() in.

File: code/game/objects/structures/tables_racks.dm [hunk 3]
```diff
@@ -544,11 +544,12 @@
...
 
 /obj/structure/table/optable/proc/check_patient()
-	var/mob/living/carbon/human/M = locate(/mob/living/carbon/human, loc)
+	var/mob/living/carbon/M = locate(/mob/living/carbon) in loc
 	if(M)
 		if(M.resting)
 			patient = M
-			return TRUE
+			if(ishuman(M) ||  ismonkey(M))
+				return TRUE
```
Comment (Rohesie) [inline code/game/objects/structures/tables_racks.dm:552]: This logic is a bit odd, as it might set up the patient while returning a falsey value.

Comment (Archanial) [reply code/game/objects/structures/tables_racks.dm:552]: That's the point. You can't output simplemob data to opcomps, but you will be able to perform researched surgeries on everything.

Comment (Rohesie) [reply code/game/objects/structures/tables_racks.dm:552]: While it makes sense given the context, it's pretty confusing logic.
This proc doubles as a human-or-monkey-patient check and a way to set the `patient` var.
Could you change it, or at least its name, for something more sane and clear?
Ideally you substitute it for a proc that checks, and handle the `patient` var assignment elsewhere.

**Comment by Archanial** [comment]
That should do it?

File: code/game/objects/structures/tables_racks.dm [hunk 4]
```diff
@@ -541,18 +541,22 @@
...
 			patient = M
-			if(ishuman(M) ||  ismonkey(M))
-				return TRUE
 	else
 		patient = null
-		return FALSE
+
+/obj/structure/table/optable/proc/check_eligible_patient()
+	get_patient()
+	if(patient)
```
Comment (Rohesie) [inline code/game/objects/structures/tables_racks.dm:556]: Early return.

File: code/game/objects/structures/tables_racks.dm [hunk 5]
```diff
@@ -553,10 +553,10 @@
...
 /obj/structure/table/optable/proc/check_eligible_patient()
 	get_patient()
-	if(patient)
-		if(ishuman(patient) ||  ismonkey(patient))
-			return TRUE
-	return FALSE
+	if(!patient)
+		return FALSE
+	else if(ishuman(patient) ||  ismonkey(patient))
+		return TRUE
```
Comment (Rohesie) [inline code/game/objects/structures/tables_racks.dm]:
```suggestion
	if(!patient)
		return FALSE
	if(ishuman(patient) ||  ismonkey(patient))
		return TRUE
	return FALSE
```

File: code/game/objects/structures/tables_racks.dm [hunk 6]
```diff
@@ -541,17 +541,22 @@
...
-			return TRUE
 	else
 		patient = null
+
+/obj/structure/table/optable/proc/check_eligible_patient()
+	get_patient()
+	if(!patient)
 		return FALSE
+	else if(ishuman(patient) ||  ismonkey(patient))
+		return TRUE
```
Comment (Rohesie) [inline code/game/objects/structures/tables_racks.dm:559]:
```suggestion
	if(ishuman(patient) ||  ismonkey(patient))
		return TRUE
	return FALSE

```

**Review by Rohesie — CHANGES_REQUESTED** [review]

**Review by Rohesie — APPROVED** [review]

**Pull request merged**